frontend/views/containers/chatroom/SendArea.vue

```diff
@@ -190,9 +243,76 @@ export default ({
         return this.createNewLine()
       }
     },
+    updateMentionKeyword () {
+      let value = this.$refs.textarea.value.slice(0, this.$refs.textarea.selectionStart)
+      const lastIndex = value.lastIndexOf('@')
+      const regExWordStart = /(\s)/g // RegEx Metacharacter \s
+      if (lastIndex === -1 || (lastIndex > 0 && !regExWordStart.test(value[lastIndex - 1]))) {
+        return this.endMention()
+      }
+      value = value.slice(lastIndex + 1)
+      if (regExWordStart.test(value)) {
+        return this.endMention()
+      }
+      this.startMention(value, lastIndex)
+    },
+    handleKeydown (e) {
+      if (Object.values(caretKeyCodes).includes(e.keyCode)) {
```

Comment:
This code is going to be run many times when the user is typing. So I think it would be nice to have a check without intermediary array creation here, to avoid garbage-collection of hundreds of small arrays when typing long messages.

Well maybe the engine is smart enough to not re-compute `Object.values(caretKeyCodes)` every time, but I'm not sure.

So in case it doesn't, I would try a switch statement or something like `if (caretKeyCodeValues[e.keyCode]))`, where `const caretKeyCodeValues = Object.values(caretKeyCodes)` would be defined next to `caretKeyCodes`.

Reply:
> Well maybe the engine is smart enough to not re-compute Object.values(caretKeyCodes) every time, but I'm not sure.

But I am sure. :). Nice catch, @snowteamer

Reply:
```
const caretKeyCodes = {
  ArrowLeft: 37,
  ArrowUp: 38,
  ArrowRight: 39,
  ArrowDown: 40,
  Esc: 27,
  End: 35,
  Home: 36
}
const caretKeyCodeValues = Object.fromEntries(Object.values(caretKeyCodes).map(v => [v, true]))

if (caretKeyCodeValues[e.keyCode]) {
  ...
}
```

I think this is better.

Reply:
Ok! 😄 My example code was wrong, sorry